### ISSUE TITLE
Create a specification and a base implementation for CW-4626

### ISSUE THREAD

**Issue by aesmonty** [CosmWasm/cw-plus#725]
### TLDR

The goal is to adapt the EIP-4626 standard for vaults and interest-bearing tokens to CosmWasm

### Background

The EIP-4626 (https://eips.ethereum.org/EIPS/eip-4626) is a standard for tokenized vaults, with the goal of increasing composability between DeFi apps, while providing a more secure and standardized interface. 

### Motivation

Vaults are commonly implemented in a wide range of Defi applications such as lending markets, yields aggregators, etc. These vaults are normally implemented as yield-bearing tokens and should conform with the CW20 as well. Given how widespread their use is, it would be very helpful to adopt a shared interface of this "Defi lego", to improve developer experience, forward compatibility, and security.

### Next steps

If this CW standard is deemed to be useful and we agree on a high-level spec, I can bootstrap the efforts and lead the implementation. It'd be great to do this collaboratively, so any help or contributor is more than welcome :)

**Comment by chinhnotional** [comment]
Hi, I think it is interesting and necessary to bring DeFi smart contract standards to Cosmwasm. So, we should start at CW4626.

**Comment by ethanfrey** [comment]
I see the purpose of a standard reference for such logic.

`cw-plus` is defined for a basic set of contracts that are production ready and serve as references for common actions. But is not intended to implement every use case.

We also created the [`cw-token`](https://github.com/CosmWasm/cw-token) repo for many more extensions and that would be a great place for such a token contract. If you make a PR for an implementation, I am happy to get it reviewed.

**Comment by ethanfrey** [comment]
Again, I would move this issue to the other repo. Happy to see contributions from the community there
